### DISCUSSION TITLE
Tor_internal keeps getting timed out. Am I implementing it incorrectly?

### DISCUSSION THREAD

**Discussion by armandoanzellini** [scholarly-python-package/scholarly#288] [Q&A] [answered]
Hi all, I have some code trying to use Tor_internal as my proxy. I've implemented it as below:

    pg = ProxyGenerator()

    tor_loc = 'C:\\directory to tor exe'

    pg.Tor_Internal(tor_cmd = tor_loc)

    scholarly.use_proxy(pg)

But whenever I run that, Tor opens a browser with two open tabs: https://__reloadtorrconsighup/ and https://0.0.0.0/

However, I click "try again" and it just repeats the same error. When I close the browser I get the following Traceback:

    Traceback (most recent call last):

      File "<ipython-input-5-6f6a617ff629>", line 5, in <module>
        pg.Tor_Internal(tor_cmd = tor_loc)

      File "C:\Users\Armando\Anaconda3\lib\site-packages\scholarly\_proxy_generator.py", line 249, in Tor_Internal
        self._tor_process = stem.process.launch_tor_with_config(

      File "C:\Users\Armando\Anaconda3\lib\site-packages\stem\process.py", line 296, in launch_tor_with_config
        return launch_tor(tor_cmd, args, torrc_path, completion_percent, init_msg_handler, timeout, take_ownership)

      File "C:\Users\Armando\Anaconda3\lib\site-packages\stem\process.py", line 158, in launch_tor
        raise OSError('Process terminated: %s' % last_problem)

     OSError: Process terminated: Timed out.

Can someone help? What am I missing?

Thanks in advance!

**Comment by ipeirotis** [comment]
Seems to be an issue with your Tor installation. You need to give permissions to Tor to write to your filesystem, especially the location where the torrc  is stored

**Comment by armandoanzellini** [comment, marked as answer]
After messing around with it, I figured out it was two problems. The first was the permissions, as ipeirotis mentioned, but the second was that I had implemented it incorrectly. I was pointing to the firefox.exe file not the tor.exe file. My bad, but now fixed.
